### ISSUE TITLE
How to use a module that exports a function instead of class with a constructor?

### ISSUE THREAD

**Issue by askdesigners** [zazoomauro/node-dependency-injection#138]
I'm trying to wire up some mongoose hooks as a dependency so I can test them. Since they are heavily dependent on using `this` to modify the current document, it becomes really tricky to have those inside of a class. 

Is there a way to inject a service without the library trying to instantiate it with a constructor call?

**Comment by zazoomauro** [comment]
@askdesigners sorry about the late response.
Can you please give me an example of your usage please?

**Comment by zazoomauro** [comment]
@askdesigners can we close this issue?

**Comment by askdesigners** [comment]
@zazoomauro ah crap sorry just seeing this now!

So like I said, we're wanting to put mongoose hooks into a module that we can pass around with the DI library. 

The way the hooks framework is built, it relies on modifying the `this` reference to change things, or make things happen before a database save. If we put all the hooks into a class then `this` becomes a class reference, rather than the document reference that we want it to be. 

For instance this snippet
```
  User.schema.pre("save", async function (next) {
    if (this.isNew && this.type === "client") {
      try {
        await createProfileForUser(this);
      } catch (error) {
        logger.error({
          message: "[ USER CREATE HOOK ] Failed to make profile for user",
          error,
          client: this.email,
        });
      }
    }
    next();
  });
```
You can see here that `this.isNew` is a reference to the DB object, rather than the class the function is a part of. 

Basically what I'm wondering is if it's possible to have a pure function, rather than a class, as an injectable module in the library.
